core/src/main/scala/chisel3/probe/package.scala

```diff
@@ -53,10 +53,21 @@ package object probe extends SourceInfoDoc {
     clone.bind(OpBinding(Builder.forcedUserModule, Builder.currentWhen))
     clone.setRef(ProbeRead(source.ref))
     // return a non-probe type Data that can be used in Data connects
-    clone.probeInfo = None
+    clearProbeInfo(clone)
     clone
   }
 
+  /** Recursively clear ProbeInfo */
+  private def clearProbeInfo[T <: Data](data: T): Unit = {
+    data match {
+      case a: Aggregate => {
+        a.probeInfo = None
+        a.elementsIterator.foreach(x => clearProbeInfo(x))
+      }
+      case leaf => { leaf.probeInfo = None }
+    }
```

Comment:
Stylistically, you do not need curly braces for the blocks after a case, also, its good to make mutating statements obvious on their own line rather than looking like an expression

I would have written:
```scala
     data match {
       case a: Aggregate =>
         a.probeInfo = None
         a.elementsIterator.foreach(x => clearProbeInfo(x))
       case leaf =>
         leaf.probeInfo = None
     }
```

You could have also probably gotten away with just:
```scala
     data.probeInfo = None
     data match {
       case a: Aggregate =>
         a.elementsIterator.foreach(x => clearProbeInfo(x))
       case _ => // Do nothing
     }
```